src/integrations/iron_handlers.rs

```diff
@@ -77,59 +85,74 @@ impl<'a, CtxFactory, Query, Mutation, CtxT>
     }
 
     fn handle_post(&self, req: &mut Request) -> IronResult<Response> {
-        let json_data = itry!(Json::from_reader(&mut req.body));
-
-        let json_obj = match json_data {
-            Json::Object(o) => o,
-            _ => return Ok(Response::with((status::BadRequest, "No JSON object was decoded"))),
-        };
-
-        let mut query = None;
-        let mut variables = Variables::new();
-
-        for (k, v) in json_obj {
-            if k == "query" {
-                query = v.as_string().map(|s| s.to_owned());
+        let mut request_payload = String::new();
+        itry!(req.body.read_to_string(&mut request_payload));
+        let json_data =
```

Comment:
This section can be simplified by using serde's derive macro for deserialization instead of manually pulling out values, e.g., https://github.com/zaeleus/juniper/blob/a321204cfc8d378fe3d8d9af2c817476cba1211d/examples/rocket/src/graphql.rs#L32-L67